File: apps/ios/Exponential/UI/GettingStarted/GettingStartedCopy.swift
import Foundation

/// Every string the getting-started checklist renders (EXP-698 r5).
///
/// One place per platform, byte-identical across all four: web's
/// `getting-started-copy.ts`, the IDE's `getting_started::copy`, Android's
/// `GettingStartedCopy.kt` and this file. A drift test
/// (`apps/web/src/components/getting-started/getting-started-copy.test.ts`)
/// reads the native sources and asserts each value appears verbatim, so every
/// literal below is a PLAIN double-quoted single-line string — no
/// interpolation, no concatenation, no escapes.
///
/// Mobile carries seven of the ten entries: the widget, helpdesk and MCP steps
/// are web/IDE-only surfaces, so the phone neither shows them nor counts them.
enum GettingStartedCopy {
    static let desktopTitle = "Get the desktop app"
    static let desktopDescription = "Runs coding sessions on your machine and registers it as one of your devices."
    static let desktopAction = "Download the desktop app"

    static let githubTitle = "Connect a GitHub repo"
    static let githubDescription = "Boards attach repositories; pull requests and coding sessions flow back into issues."
    static let githubAction = "Connect GitHub"

    static let inviteTitle = "Invite your team"
    static let inviteDescription = "Teammates share boards, reviews, and the support inbox."
    static let inviteAction = "Invite in team settings"

    static let boardTitle = "Create a board"
    static let boardDescription = "Boards hold your issues; connect a repository to code on one."
    static let boardAction = "Create a board"

    static let codingTitle = "Start coding with an agent"
    static let codingDescription = "Start coding on an issue hands it to your agent, which plans, implements, and opens the PR."
    static let codingAction = "Open Devices"

    static let actionTitle = "Create an action"
    static let actionDescription = "Reusable agent runs for your team, written by your agent from a description."
    static let actionAction = "New action"

    static let serverTitle = "Set up a server"
    static let serverDescription = "Run the headless daemon on an always-on machine to take remote Start coding requests."
    static let serverAction = "Copy install command"

    /// The section header and its progress readout.
    static let sectionTitle = "Getting started"

    static func progress(done: Int, total: Int) -> String {
        "\(done)/\(total) done"
    }
}

/// One checklist step. The order of the cases IS the display order, minus the
/// three web-only entries (widget, helpdesk, mcp).
enum GettingStartedEntryKey: String, CaseIterable, Identifiable {
    case desktop
    case github
    case invite
    case board
    case coding
    case action
    case server

    var id: String { rawValue }

    var title: String {
        switch self {
        case .desktop: GettingStartedCopy.desktopTitle
        case .github: GettingStartedCopy.githubTitle
        case .invite: GettingStartedCopy.inviteTitle
        case .board: GettingStartedCopy.boardTitle
        case .coding: GettingStartedCopy.codingTitle
        case .action: GettingStartedCopy.actionTitle
        case .server: GettingStartedCopy.serverTitle
        }
    }

    var entryDescription: String {
        switch self {
        case .desktop: GettingStartedCopy.desktopDescription
        case .github: GettingStartedCopy.githubDescription
        case .invite: GettingStartedCopy.inviteDescription
        case .board: GettingStartedCopy.boardDescription
        case .coding: GettingStartedCopy.codingDescription
        case .action: GettingStartedCopy.actionDescription
        case .server: GettingStartedCopy.serverDescription
        }
    }

    var actionLabel: String {
        switch self {
        case .desktop: GettingStartedCopy.desktopAction
        case .github: GettingStartedCopy.githubAction
        case .invite: GettingStartedCopy.inviteAction
        case .board: GettingStartedCopy.boardAction
        case .coding: GettingStartedCopy.codingAction
        case .action: GettingStartedCopy.actionAction
        case .server: GettingStartedCopy.serverAction
        }
    }
}

/// One-line hints for locked entries, keyed by entry + the step that unlocks
/// it. Same wording as web's `lockedHint`, minus the widget arm.
func gettingStartedLockedHint(
    entry: GettingStartedEntryKey,
    lockedBy: GettingStartedEntryKey
) -> String {
    if entry == .coding && lockedBy == .desktop {
        return "Connect a machine first — coding sessions run on the desktop app or a registered server."
    }
    if entry == .coding && lockedBy == .github {
        return "Connect a GitHub repo first. Coding sessions need a repo-backed board."
    }
    if entry == .coding && lockedBy == .board {
        return "Create a board with a repository first."
    }
    if entry == .action {
        return "Connect a machine first — the action creator runs on the desktop app or a registered server."
    }
    return "Complete \"\(lockedBy.title)\" first."
}

File: apps/android/app/src/main/java/com/exponential/app/ui/gettingstarted/GettingStartedCopy.kt
package com.exponential.app.ui.gettingstarted

/**
 * The getting-started checklist's words, byte-identical to the other three
 * clients (EXP-698 r5, Mechanism B). Web owns the canonical table in
 * `components/getting-started/getting-started-copy.ts` and its
 * `getting-started-copy.test.ts` reads THIS file, asserting every string
 * appears here verbatim — so each value must stay a plain, single-line,
 * double-quoted literal with no escapes and no non-ASCII punctuation.
 *
 * Mobile carries seven of the ten entries: the widget, helpdesk and MCP steps
 * are set up on web or the IDE, and pointing a phone at them would be a step
 * nobody can finish where they are standing.
 */
object GettingStartedCopy {
    const val DESKTOP_TITLE = "Get the desktop app"
    const val DESKTOP_DESCRIPTION = "Runs coding sessions on your machine and registers it as one of your devices."
    const val DESKTOP_ACTION = "Download the desktop app"

    const val GITHUB_TITLE = "Connect a GitHub repo"
    const val GITHUB_DESCRIPTION = "Boards attach repositories; pull requests and coding sessions flow back into issues."
    const val GITHUB_ACTION = "Connect GitHub"

    const val INVITE_TITLE = "Invite your team"
    const val INVITE_DESCRIPTION = "Teammates share boards, reviews, and the support inbox."
    const val INVITE_ACTION = "Invite in team settings"

    const val BOARD_TITLE = "Create a board"
    const val BOARD_DESCRIPTION = "Boards hold your issues; connect a repository to code on one."
    const val BOARD_ACTION = "Create a board"

    const val CODING_TITLE = "Start coding with an agent"
    const val CODING_DESCRIPTION = "Start coding on an issue hands it to your agent, which plans, implements, and opens the PR."
    const val CODING_ACTION = "Open Devices"

    const val ACTION_TITLE = "Create an action"
    const val ACTION_DESCRIPTION = "Reusable agent runs for your team, written by your agent from a description."
    const val ACTION_ACTION = "New action"

    const val SERVER_TITLE = "Set up a server"
    const val SERVER_DESCRIPTION = "Run the headless daemon on an always-on machine to take remote Start coding requests."
    const val SERVER_ACTION = "Copy install command"

    /** The section's own chrome, shared with web + the IDE. */
    const val SECTION_TITLE = "Getting started"

    fun title(key: GettingStartedEntryKey): String = when (key) {
        GettingStartedEntryKey.Desktop -> DESKTOP_TITLE
        GettingStartedEntryKey.Github -> GITHUB_TITLE
        GettingStartedEntryKey.Invite -> INVITE_TITLE
        GettingStartedEntryKey.Board -> BOARD_TITLE
        GettingStartedEntryKey.Coding -> CODING_TITLE
        GettingStartedEntryKey.Action -> ACTION_TITLE
        GettingStartedEntryKey.Server -> SERVER_TITLE
    }

    fun description(key: GettingStartedEntryKey): String = when (key) {
        GettingStartedEntryKey.Desktop -> DESKTOP_DESCRIPTION
        GettingStartedEntryKey.Github -> GITHUB_DESCRIPTION
        GettingStartedEntryKey.Invite -> INVITE_DESCRIPTION
        GettingStartedEntryKey.Board -> BOARD_DESCRIPTION
        GettingStartedEntryKey.Coding -> CODING_DESCRIPTION
        GettingStartedEntryKey.Action -> ACTION_DESCRIPTION
        GettingStartedEntryKey.Server -> SERVER_DESCRIPTION
    }

    fun action(key: GettingStartedEntryKey): String = when (key) {
        GettingStartedEntryKey.Desktop -> DESKTOP_ACTION
        GettingStartedEntryKey.Github -> GITHUB_ACTION
        GettingStartedEntryKey.Invite -> INVITE_ACTION
        GettingStartedEntryKey.Board -> BOARD_ACTION
        GettingStartedEntryKey.Coding -> CODING_ACTION
        GettingStartedEntryKey.Action -> ACTION_ACTION
        GettingStartedEntryKey.Server -> SERVER_ACTION
    }

    /**
     * One-line hint for a locked entry, keyed by the entry plus the step that
     * unlocks it. Unchanged wording from web's `lockedHint` (only the two
     * lockable mobile entries can reach here).
     */
    fun lockedHint(entry: GettingStartedEntryKey, lockedBy: GettingStartedEntryKey): String = when {
        entry == GettingStartedEntryKey.Coding && lockedBy == GettingStartedEntryKey.Desktop ->
            "Connect a machine first — coding sessions run on the desktop app or a registered server."
        entry == GettingStartedEntryKey.Coding && lockedBy == GettingStartedEntryKey.Github ->
            "Connect a GitHub repo first. Coding sessions need a repo-backed board."
        entry == GettingStartedEntryKey.Coding && lockedBy == GettingStartedEntryKey.Board ->
            "Create a board with a repository first."
        entry == GettingStartedEntryKey.Action ->
            "Connect a machine first — the action creator runs on the desktop app or a registered server."
        else -> "Complete \"${title(lockedBy)}\" first."
    }
}
